### PR TITLE
Exclude the 'Locals' section if there are no locals to display

### LINKED ISSUE #37: Exclude the “Locals” section if there are no locals defined
If there are no locals (besides `self`, `msg`, and `note`), we should hide the “Locals” section of the output.

### PR DESCRIPTION
**Pull request by leifwalsh**
Closes #37

### CODE REVIEW

File: marbles/core/marbles/core/marbles.py [hunk 1]
```diff
@@ -253,15 +254,21 @@ def note(self):
 
     @property
     def locals(self):
-        '''Returns a string displaying the public (a.k.a., not internal
```
Comment (leifwalsh) [inline marbles/core/marbles/core/marbles.py]: oops this docstring has been wrong for a while

File: marbles/core/tests/test_marbles.py [hunk 2]
```diff
@@ -677,7 +698,7 @@ def test_exclude_ignored_locals(self):
         with self.assertRaises(ContextualAssertionError) as ar:
             self.case.test_locals()
         e = ar.exception
-        locals_section = e._format_locals(e.locals).split('\n')
```
Comment (leifwalsh) [inline marbles/core/tests/test_marbles.py]: This test is a little brittle

File: marbles/core/marbles/core/marbles.py (hunk 1 above)
Comment (thejunglejane) [reply marbles/core/marbles/core/marbles.py]: whoops

File: marbles/core/marbles/core/marbles.py [hunk 3]
```diff
@@ -253,15 +254,21 @@ def note(self):
...
-        or name-mangled) locals defined within the test.
+        '''A dict containing the locals defined within the test.'''
+        return self._locals
+
+    @property
+    def displayed_locals(self):
```
[the remark this answers is not in the document]
Comment (leifwalsh) [reply marbles/core/marbles/core/marbles.py]: Done

File: marbles/core/tests/test_marbles.py [hunk 4]
```diff
@@ -427,6 +427,27 @@ def test_missing_annotation(self):
             with self.assertRaises(AssertionError):
                 self.case.test_missing_annotation_fail()
 
+    def test_locals_hidden_when_missing(self):
```
[the remark this answers is not in the document]
Comment (leifwalsh) [reply marbles/core/tests/test_marbles.py]: Done, and yes, we should re-architect them soon.

**Comment by thejunglejane** [comment]
Ah crap I shouldn't have pressed the "update branch" button, sorry.

**Comment by thejunglejane** [comment]
I hate merge commits ))):

**Pull request merged**